src/main/java/com/webapp/TextBook/sharedFiles/VerifySessionUser.java

```diff
@@ -0,0 +1,70 @@
+package com.webapp.TextBook.sharedFiles;
+
+import com.webapp.TextBook.repository.data_access.User;
+import com.webapp.TextBook.viewModel.sharedViewModel.loginUserInfo.LoginUserInfo;
+import org.javatuples.Triplet;
+
+import javax.validation.constraints.NotNull;
+import java.util.Optional;
+
+public class VerifySessionUser {
+    //used on every api and web controller call that is NOT
+    // the login page to verify the user has been validated prior.
+    // This bar man in the middle attacks and injection
+    private static final String ERROR_SESSION_USER_MESSAGE =
+            "User state not set or does not match session. Please reset the session by logging in. Not your problem? Please contact an IT Support member.";
+
+    private static final String ERROR_SESSION_USER_UNSET = "Session User has not been set";
+
+    public static Triplet<Boolean, String, Optional<User>> isSessionUserValid(@NotNull final LoginUserInfo user){
+
+        if(user == null){
+            return  new Triplet<Boolean, String, Optional<User>>(
+                    false,
+                    ERROR_SESSION_USER_MESSAGE,
+                    Optional.empty());
+        }
+
+        // wild card capture guaranteed -- type parameter will be User
+        final Optional<User> userOptional = VerifySessionUser.captureWildCardUser(
+                SharedSessionData.getSessionState(SharedSessionData.USER_KEY));
+
+        if(userOptional.isEmpty())
```

Comment:
what does it mean is user optional is empty?
will it ever happened -- look at the method in shared session data for the answer!